dotcom-rendering/src/web/browser/islands/doHydration.ts

```diff
@@ -14,6 +14,11 @@ import { initPerf } from '../initPerf';
  * @param element The location on the DOM where the component to hydrate exists
  */
 export const doHydration = (name: string, data: any, element: HTMLElement) => {
+	// If this function has already been run for an element then don't try to
+	// run it a second time
+	const alreadyHydrated = element.dataset.guReady;
```

Comment:
If this property returns a `boolean` how do we feel about naming convention `isAlreadyHydrated`?

Reply:
Good question!

It's related to [this discussion](https://github.com/guardian/dotcom-rendering/pull/3865#discussion_r791032527) on another PR. We refer to the act of 'hydration' a lot and it's accurate inasmuch as we are running the _hydrate_ function here, always. But in truth there are two contexts

1) Some content was server side rendered and we're 'hydrating' it (making it interactive)
2) Nothing was server side rendered and we're simply 'rendering' content here

But we use the `React.hydrate` function in both cases because in truth this function will fallback to rendering the content if it can't hydrate it so it's semantically correct to call this hydration but perhaps confusing to say it _has been_ hydrated.

The purpose of the data attribute is so we can know when this function has completed to 'gu-ready` makes more sense